### PR TITLE
chore: parametrize Rockcraft channel and clean up rocks CI

### PR DESCRIPTION
**Pull request by dariofaccin**
This PR allows to parametrize rockcraft channel in inputs and cleans the rocks CI actions. The total saved time is around 30%.

This PR completely removes the setup environment step with concierge, since neither K8s or Juju are required. In addition, debug steps for Juju, Charmcraft, K8s, are removed since these snaps are not even installed.

This PR also removes the wrong `required: true` when a default is defined. This is an intermediate step required to fully remove microk8s from our CI: this option will be removed in rocks repositories, and then finally removed from here with another PR.

This PR also fixes an issue in `chaci` rock integration for which dry-run integration PRs would be always created against `main` and actual integration PRs are failing when the source branch is in the form of `track/*` (or more in general contains multiple `/`). Example of failure can be found [here](https://github.com/canonical/kubeflow-trainer-rocks/actions/runs/26560370495/job/78243617645).

<details>
<summary>New dry-run</summary>

```
[DRY-RUN] Would commit to branch 'integrate-kubeflow-trainer-2.1.0-8483ba9' with message: chore: integrate rock image kubeflow-trainer:2.1.0-8483ba9
[DRY-RUN] Would open PR: integrate-kubeflow-trainer-2.1.0-8483ba9 → track/2.1
```
</details>

This PR also bumps the version of many actions to use Node24.

This PR has been tested [here](https://github.com/canonical/kubeflow-trainer-rocks/actions/runs/26574292882).

### CODE REVIEW

File: .github/workflows/integration-test-rock.yaml [hunk 1]
```diff
@@ -79,16 +85,6 @@ jobs:
...
 
-      - name: Install dependencies
-        run: pipx install tox
-
-      - name: Setup environment
-        run: |
-          sudo apt-get remove -y docker-ce docker-ce-cli containerd.io
-          sudo rm -rf /run/containerd
-          sudo snap install concierge --classic
-          sudo concierge prepare --trace
```
Comment (MattiaSarti) [inline .github/workflows/integration-test-rock.yaml:90]: question: why are we undoing this? Some rock repositories may not install tools themselves in their CI

File: .github/workflows/integration-test-rock.yaml [hunk 2]
```diff
@@ -98,41 +94,3 @@ jobs:
...
-          kubectl describe gateways -A | tee tmp/kube-gateways.txt
-          kubectl describe deployments -A | tee tmp/kube-deployments.txt
-          kubectl describe replicasets -A | tee tmp/kubectl-replicasets.txt
-
-      - name: Upload debug artifacts
-        if: failure() || cancelled()
-        uses: actions/upload-artifact@v4
-        with:
-          name: test-run-artifacts
-          path: tmp
```
Comment (MattiaSarti) [inline .github/workflows/integration-test-rock.yaml:138]: question: are we sure that without this proper debugging information is collected for all rock repositories?

File: .github/workflows/integration-test-rock.yaml (hunk 1 above)
Comment (dariofaccin) [reply .github/workflows/integration-test-rock.yaml:90]: We don't need K8s, Charmcraft, Juju for rock integration. Regarding `tox`, it's already installed in _Install dependencies_, few steps above.

File: .github/workflows/integration-test-rock.yaml (hunk 2 above)
Comment (dariofaccin) [reply .github/workflows/integration-test-rock.yaml:138]: These logs are fetched from Charmcraft, Juju and K8s: rock integrations do not use these.

File: .github/workflows/build-rock.yaml [hunk 3]
```diff
@@ -43,9 +43,10 @@ jobs:
         uses: jlumbroso/free-disk-space@54081f138730dfa15788a46383842cd2f914a1be
 
       - name: Checkout repository
-        uses: actions/checkout@v4
+        uses: actions/checkout@v6
         with:
           ref: ${{ inputs.source-branch }}
+          persist-credentials: false
```
Comment (mvlassis) [inline .github/workflows/build-rock.yaml:49]: question: Is there a *specific* reason we set this here? Is it for security reasons? Is it related to the [changes in v6](https://github.com/actions/checkout#whats-new)?

File: .github/workflows/integration-test-rock.yaml (hunk 1 above)
Comment (mvlassis) [reply .github/workflows/integration-test-rock.yaml:90]: This action is never actually useful, right? It tries to run the integration tests in rock repos, which should always be empty. I found [an example run](https://github.com/canonical/pipelines-rocks/actions/runs/26157620917/job/76942154734) from `pipelines-rocks`. Indeed it doesn't need the setup, although maybe we should remove it altogether...

**Review by mvlassis — COMMENTED** [review]
Left a few small nitpicks

File: .github/workflows/build-rock.yaml (hunk 3 above)
Comment (dariofaccin) [reply .github/workflows/build-rock.yaml:49]: With `v6` I faced [this issue](https://github.com/actions/checkout/issues/2299) whenever there are more nested check-outs. In the link there's also the commit showing that an authorization header is now added if `persist-credentials` is enabled.

File: .github/workflows/integration-test-rock.yaml (hunk 1 above)
Comment (dariofaccin) [reply .github/workflows/integration-test-rock.yaml:90]: It's true that currently it is a co-op (actually no-test), but it only consumes 1 minute and it's already there. If we remove this and in future we decide to add integration tests for rocks, it would be duplicate effort.

Comment (MattiaSarti) [reply .github/workflows/integration-test-rock.yaml:90]: > We don't need K8s, Charmcraft, Juju for rock integration. Regarding tox, it's already installed in Install dependencies, few steps above.

We don't need them as far as we run the simplified versions of rock integration tests that we are running at the moment, but I find it reasonable to make it lighter if no rock repository is using that. In case, we'll simply reintroduce this in the future, when and if things change. So I'm fine with this removal.

File: .github/workflows/integration-test-rock.yaml (hunk 2 above)
Comment (MattiaSarti) [reply .github/workflows/integration-test-rock.yaml:138]: This goes hand in hand with my comment above.

**Review by MattiaSarti — APPROVED** [review]

**Comment by misohu** [comment]
The reason we have there the k8s setup and juju is that originally the idea was that as part of the rock ci we will run integration tests of the charm where the rock is integrated ... instead of removing completely maybe we should just parametrise this step to be off  by default ?

**Review by misohu — APPROVED** [review]

**Review by MattiaSarti — APPROVED** [review]
